src/ssr-template-helper.js

```diff
@@ -113,34 +107,40 @@ export class SsrTemplateHelper {
    */
   buildPayload_(
     request, mustacheTemplate, opt_templates, opt_attributes = {}) {
-    const ampComponent = dict({
-      'type': this.sourceComponent_,
-      'successTemplate': {
+    const ampComponent = dict({'type': this.sourceComponent_});
+
+    const successTemplate = (opt_templates && opt_templates['successTemplate'])
```

Comment:
If you pass in an actual template object to `mustacheTemplate` (which can probably be better-named, since currently there's `mustacheTemplate` and `opt_templates` which is kinda confusing), then you can just do:

```js
const successTemplate = (opt_templates && opt_templates['successTemplate'])
    ? opt_templates['successTemplate'] : mustacheTemplate;

if (successTemplate) {
  ...
   'payload': successTemplate.innerHTML,
  ...
}
```

Reply:
Hmmm does it really work? mustacheTemplate (or ampListTemplate now) can also be null.

Reply:
If mustacheTemplate (or ampListTemplate now) is null, it will fail the `if (successTemplate)` check, wouldn't it? I'm not sure if you need all those conditionals... seems redundant

Reply:
I see what you mean, yes this works, done.